### ISSUE TITLE
plugin autoInitialization

### ISSUE THREAD

**Issue by timomayer** [virtualidentityag/frontend-framework#30]
Instead of initializing each plugin manually in main.js, place a file called init.js/ts in the component folder with something in like:

{"conditional": false, "init": "function(){ // init code}"}
we can use this with our new handlebars json file loader to place it inside a data-init="" attribute for conditional Loading

and (with an additional build framework task) we can collect all the init files from components folder, check if conditional is false, if yes generate and "auto init block" and inject it into main.js (use a simple string replace for that with a placeholder like // ###autoInitComponents###

@janrembold @alxbenz what do you think?

**Comment by janrembold** [comment]
Yes, I already thought about an approach like that. But instead of a placeholder i thought about a handlebars helper that includes the loader attributes if conditional: true or ignores it if not.

**Comment by timomayer** [comment]
Yes thats great for the data-resource I thought we can use the placeholder in main.js to place things there which are not conditionally loaded

**Comment by janrembold** [comment]
To fully automate this shouldn't we collect all component configs, parse the loading mechanism and if

- conditional: true
=> let the handlebars helper add resources and init to the components markup

- conditional: false
=> add resources to global script file (with useref or its future equivalent) and concatenate init scripts into main.js

What do you think?
